docs/julia.rst

```diff
@@ -0,0 +1,340 @@
+| or256(x:256, y:256) -> z:256                | bitwise or of x and y                                           |
++---------------------------------------------------------------------------------------------------------------+
+| xor256(x:256, y:256) -> z:256               | bitwise xor of x and y                                          |
++---------------------------------------------------------------------------------------------------------------+
+| shl256(x:256, y:256) -> z:256               | logical left shift of x by y                                    |
++---------------------------------------------------------------------------------------------------------------+
+| shr256(x:256, y:256) -> z:256               | logical right shift of x by y                                   |
++---------------------------------------------------------------------------------------------------------------+
+| sar256(x:256, y:256) -> z:256               | arithmetic right shift of x by y                                |
++---------------------------------------------------------------------------------------------------------------+
+| byte(n:256, x:256) -> v:256                 |  nth byte of x, where the most significant byte is the 0th byte |
+| Cannot this be just replaced by and256(shr(n, x), 0xff) and let it be optimised out by the EVM backend?       |
++---------------------------------------------------------------------------------------------------------------+
+| *Memory and storage*                                                                                          |
++---------------------------------------------------------------------------------------------------------------+
+| mload(p) -> v:256                           | mem[p..(p+32))                                                  |
++---------------------------------------------------------------------------------------------------------------+
+| mstore(p, v:256)                            | mem[p..(p+32)) := v                                             |
++---------------------------------------------------------------------------------------------------------------+
+| mstore8(p, v:256)                           | mem[p] := v & 0xff    - only modifies a single byte             |
++---------------------------------------------------------------------------------------------------------------+
+| sload(p) -> v:256                           | storage[p]                                                      |
++---------------------------------------------------------------------------------------------------------------+
+| sstore(p, v:256)                            | storage[p] := v                                                 |
++---------------------------------------------------------------------------------------------------------------+
+| msize() -> size:256                         | size of memory, i.e. largest accessed memory index              |
```

Comment:
Somehow `MSIZE` in EVM always returns multiples of 32.

Maybe `the smallest word-aligned memory index bigger than any accessed memory index` (but this phrase is a bit mouthful).